Frameworks/UIKit/UIApplicationMain.mm

```diff
@@ -166,18 +166,15 @@ int UIApplicationMainInit(NSString* principalClassName,
     if (infoDict != nil) {
         NSArray* fonts = [infoDict objectForKey:@"UIAppFonts"];
         if (fonts != nil) {
+            NSMutableArray* fontURLs = [NSMutableArray array];
             for (NSString* curFontName in fonts) {
-                NSString* path = [[NSBundle mainBundle] pathForResource:curFontName ofType:nil];
-                if (path != nil) {
-                    NSData* data = [NSData dataWithContentsOfFile:path];
-                    if (data != nil) {
-                        UIFont* font = [UIFont fontWithData:data];
-                        if (font != nil) {
-                            CTFontManagerRegisterGraphicsFont((CGFontRef)font, NULL);
-                        }
-                    }
+                NSURL* url = [[NSBundle mainBundle] URLForResource:curFontName withExtension:nil];
```

Comment:
I don't think it can; curFontName is pulled from info.plist with its extension fully formed.

`... URLForResource:@"x.ttf" withExtension:@"ttf"` will look for a file x.ttf.ttf.

---
In reply to: [89406546](https://github.com/Microsoft/WinObjC/pull/1454#discussion_r89406546) [](ancestors = 89406546)